src/views/router_views/enter-your-details/enter-your-details.njk

```diff
@@ -0,0 +1,146 @@
+{% extends "layouts/default.njk" %}
+
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/select/macro.njk" import govukSelect %}
+
+{% block main_content %}
+{% set premisesField = ""%}
+{% set addressLine1Field = ""%}
+{% set townOrCityField = ""%}
+{% set countryField = ""%}
+{% set postCodeField = ""%}
+    {% if errors %}
+        {% for error in errors %}
+            {% if error.path === "premises" %}
+                {% set premisesField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "addressLine1" %}
+                {% set addressLine1Field = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "addressLine2" %}
+                {% set addressLine2Field = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "townOrCity" %}
+                {% set townOrCityField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "county" %}
+                {% set countyField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "country" %}
+                {% set countryField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "postCode" %}
+                {% set postCodeField = {
+                    text: error.msg
+                }%}
+            {% endif %}
+        {% endfor %}
+        {% else %}
+            {% set errorMessage = undefined %}
+    {% endif %}
+        <form class="form" action="{{Urls.ENTER_YOUR_DETAILS}}" method="POST">
+            <fieldset class="govuk-fieldset">
+                <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+                    <h1 class="govuk-fieldset__heading">
+                        {{ title }}
+                    </h1>
+                </legend>
+                <p class="govuk-hint">We may use this to contact you about your account.</p>
+                {{ govukInput({
+                        label: {
+                            text: "Property name or number",
+                            classes: "govuk-label"
+                        },
+                        hint: {
+                            text: "For example, 'The Mill', '116' or 'Flat 37a'."
+                        },
+                        id: "premises",
+                        name: "premises",
+                        value: address.premises,
+                        classes: "{% if premisesField !== '' ? 'govuk-input--error' else '' %}",
```

Comment:
You don't need to manually add this class.
If the errorMessage field is defined then it will be added automatically.